docs/file-transfer.md

```diff
@@ -0,0 +1,97 @@
+command is still hanging, then something unexpected is going on, and I
+recommend filing an issue about it. If they are not connected, then let's try
+and debug why. (Note: you can skip to 'Manually connecting node A to node B' if
+you just want things to work. Going through the debugging process and reporting
+what happened to the ipfs team on IRC is helpful to us to understand common
+pitfalls that people run into)
+
+### Checking providers
+When requesting content on ipfs, nodes search the DHT for 'provider records' to
+see who has what content. Let's manually do that on node B to make sure that
+node B is able to determine that node A has the data. Run `ipfs dht findprovs
+<hash>`. We expect to see the peer ID of node A printed out. If this command
+returns nothing (or returns IDs that are not node A), then no record of A
+having the data exists on the network. This can happen if the data is added
+while node A does not have a daemon running. If this happens, you can run `ipfs
+dht provide <hash>` on node A to announce to the network that you have that
+hash. Then if you restart the `ipfs get` command, node B should now be able
+to tell that node A has the content it wants. If node A's peer ID showed up in
+the initial `findprovs` call, or manually providing the hash didn't resolve the
+problem, then it's likely that node B is unable to make a connection to node A.
+
+### Checking addresses
+
+In the case where node B simply cannot form a connection to node A, despite
+knowing that it needs to, the likely culprit is a bad NAT. When node B learns
+that it needs to connect to node A, it checks the DHT for addresses for node A,
+and then starts trying to connect to them. We can check those addresses by
+running `ipfs dht findpeer <node A peerID>` on node B. This command should
+return a list of addresses for node A. If it doesnt return any addresses, then
+you should try running the manual providing command from the previous steps. 
+Example output of addresses might look something like this:
+
+```
+/ip4/127.0.0.1/tcp/4001
+/ip4/192.168.2.133/tcp/4001
+/ip4/88.157.217.196/tcp/63674
+```
+
+In this case, we can see a localhost (127.0.0.1) address, a LAN address (the
+192.168.*.* one) and another address. If this third address matches your
+external IP, then the network knows a valid external address for your node. At
+this point, its safe to assume that your node has a difficult to traverse NAT
+situation. If this is the case, you can try to enable upnp or NAT-PMP on the
+router of node A and retry the process. Otherwise, you can try manually
+connecting node A to node B.
+
+### Manually connecting node A to B On node B, run `ipfs id` and take one of
+the multiaddrs that contains its public ip address, and then on node A run
+`ipfs swarm connect <multiaddr>`.  If that *still* doesn't work, then you
+should either join IRC and ask for help there, or file an issue on github.
```

Comment:
Some note on p2p-circuit may be helpful too

Reply:
@magik6k @vyzo mind suggesting some text for how to dial a circuit relay address here? (i'm not sure if you just `ipfs swarm connect /p2p-circuit/QmFooBar` or what)

Reply:
I think it'd be enough to just link https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#circuit-relay